### PR TITLE
Base Site Functionality

### PR DESCRIPTION
**Pull request by chiefpansancolt**

### CODE REVIEW

File: app/models/testimonial.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class Testimonial < ApplicationRecord
+  validates :name, :quote, :status, presence: true
+
+  scope :visable, -> { where(status: "Approved") }
```
Comment (npezza93) [inline app/models/testimonial.rb]: spelled wrong

File: app/models/slider.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class Slider < ApplicationRecord
+  has_attached_file :photo
+  validates_attachment :photo, presence: true, content_type: {
+    content_type: ["image/jpeg", "image/gif", "image/png", "image/jpg"]
+  }
+  validates :order, presence: true
+  scope :visable, -> { where(active: true).order(:order) }
```
Comment (npezza93) [inline app/models/slider.rb]: spelled wrong

File: app/models/setting.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    "https://twitter.com/" + twitter_handle
+  end
+
+  def pinterest_url
+    return unless pinterest_handle
+    "https://www.pinterest.com/" + pinterest_handle
+  end
+
+  def instagram_url
+    return unless instagram_handle
```
Comment (npezza93) [inline app/models/setting.rb]: instagram_handle.present?

File: app/models/page.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    Page.find(page_id).title
+  end
+
+  def tab_page_check
+    return unless page_type == "Tab" && page_id.nil?
+    errors.add(:page_id, "Parent Page Requried when page type is Tab")
+  end
+
+  def page_check
+    return unless page_type == "Page" && (icon.nil? || icon == "")
```
Comment (npezza93) [inline app/models/page.rb]: (icon.nil? || icon == "")  => icon.blank?

File: app/models/page.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  has_many :collections
+  has_many :markers
+  has_many :card_groups
+  has_many :contact_emails
+  has_many :photo_albums
+  has_many :exhibits
+
+  scope :active, -> { where(active: true).order(:order) }
+  scope :available_pages, -> { where(page_type: "Page").order(:order) }
+  scope :pages_tabs, -> { all.order(:order) }
```
Comment (npezza93) [inline app/models/page.rb]: just .order(:order)

**Review by npezza93 — CHANGES_REQUESTED** [review]
Seems decent. Couple tweaks. What's your code coverage sitting at?

**Comment by npezza93** [comment]
Got it 96%.

File: app/models/slider.rb (hunk 2 above)
Comment (chiefpansancolt) [reply app/models/slider.rb]: whats spelled wrong?

File: app/models/testimonial.rb (hunk 1 above)
Comment (chiefpansancolt) [reply app/models/testimonial.rb]: whats spelled wrong?

**Review by npezza93 — APPROVED** [review]

**Pull request merged**